client/app/pages/admin/Jobs.jsx

```diff
@@ -67,7 +68,12 @@ class Jobs extends React.Component {
   };
 
   render() {
-    const { isLoading, error, queueCounters, startedJobs, overallCounters, workers } = this.state;
+    const { isLoading, error, queueCounters, startedJobs, overallCounters, workers, activeTab } = this.state;
+
+    const changeTab = (newTab) => {
```

Comment:
This is the simplest way I could find that doesn't involve bringing in a router.

![giphy](https://user-images.githubusercontent.com/289488/68543933-26f1a500-03c6-11ea-8ebe-3b0b6a0ae228.gif)

Reply:
What do you mean by bringing in a router?

Reply:
Directly manipulating the location hash isn't such a good practice. The appropriate way would probably be to define routes for these sub-pages, using something like `react-router` (which might not be an entirely bad idea, as it will open the door to more routing in the app, and, subsequently, the gate to hell)

Reply:
When we get rid of Angular we should look into having a separate URL for each of them, but right now it might prove to be tricky (without reloading all the data on every URL change).

But check how we handle hash navigation on the query page (we use it to direct link to specific visualizations) and see if it's not too complex. Or is it what you already do here?

Reply:
If you're referring to [this](https://github.com/getredash/redash/blob/master/client/app/pages/queries/view.js#L518), I think it'll be overkill as the admin page is almost entirely Angular-free. @kravets-levko WDYT?

Reply:
Oh, not that monstrosity. I'm not even sure why we need it there :flushed: What you did is actually fine, let's roll with it. 

Just to make sure: `location.replace` won't make Angular trigger a reload or something?

Reply:
It doesn't trigger a page reload, but it does trigger a component remounting, which in turns fetches new information from the API (not sure if that's really a bad thing).

Reply:
I was hoping to avoid that, to keep switching tabs snappy. Otherwise we could just use separate URLs (`/admin/rq/queues`, `/admin/rq/workers`, etc) instead.

I guess that to avoid remounting, we have to move fetching one level up and pass it down to the tabs (or have a "service" class that provides this state).

Reply:
It still is snappy. It switches instantaneously, but it triggers a background reload of data, which means that when you switch to Workers, for example, you will see data immediately, and a few hundred milliseconds later that data may change.

Reply:
Oh, ok. Not ideal but let's move on.